Rakefile

```diff
@@ -224,6 +224,12 @@ task :build => ['build:all']
 ##
 ##################################################################################################
 
+# Whether this is a development or adhoc environment where we should install npm and create
+# a local database.
+def local_environment?
+  (rack_env?(:development) && !CDO.chef_managed) || rack_env?(:adhoc)
```

Comment:
doesn't hurt to be safe, but I am curious if there is there an environment with `rack_env?(:development) && CDO.chef_managed` ? If not this could be simplified.